projects/clusteringress/pkg/translator/translator_syncer.go

```
@@ -66,5 +73,66 @@ func (s *translatorSyncer) Sync(ctx context.Context, snap *v1.TranslatorSnapshot
 		return err
 	}
 
+	if err := s.propagateProxyStatus(ctx, proxy, snap.Clusteringresses); err != nil {
+		return errors.Wrapf(err, "failed to propagate proxy status "+
+			"to clusteringress objects")
+	}
+
+	return nil
+}
+
+// propagate to all clusteringresses the status of the proxy
+func (s *translatorSyncer) propagateProxyStatus(ctx context.Context, proxy *gloov1.Proxy, clusterIngresses v1alpha1.ClusterIngressList) error {
+	if proxy == nil {
+		return nil
+	}
+	timeout := time.After(time.Second * 30)
+	ticker := time.Tick(time.Second / 2)
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-timeout:
+			return errors.Errorf("timed out waiting for proxy status to be updated")
+		case <-ticker:
+			// poll the proxy for an accepted or rejected status
+			updatedProxy, err := s.proxyClient.Read(proxy.Metadata.Namespace, proxy.Metadata.Name, clients.ReadOpts{Ctx: ctx})
+			if err != nil {
+				return err
+			}
+			switch updatedProxy.Status.State {
+			case core.Status_Pending:
+				continue
+			case core.Status_Rejected:
+				contextutils.LoggerFrom(ctx).Errorf("proxy was rejected by gloo: %v",
```

If our proxy gets rejected, shouldn't we propagate this failure to the `ClusterIngress` so that it is reflected in its status?

---

if the clusteringress has already been marked as Ready, there's no way to revert it to NotReady (other than zeroing out the status i suppose). knative doesn't actively monitor the status of the clusteringress, it simply creates the CI, waits for status to be set to ready within a certain timeout, then continues

check out https://github.com/knative/serving/blob/master/pkg/reconciler/clusteringress/clusteringress.go#L217 to see how knative sets the clusteringress status for istio - there is no status update for the error case, only the success case